test/mpi/test_mpi_cuda.c

```diff
@@ -0,0 +1,668 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/debug/log_def.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <mpi.h>
+#include <cuda.h>
+#include <pthread.h>
+
+#define SIZE_S 8
+#define SIZE_M 4 * 1024 * 1024
+#define SIZE_L 16 * 1024 * 1024
+
+#define TEST_NAME_FMT      "TEST[%d:%-17s:%-17s:*       ]"
+#define TEST_NAME_SIZE_FMT "TEST[%d:%-17s:%-17s:%-8zi]"
+
+#define PRINT_ROOT(fmt, ...) \
+    do { \
+        int _rank; \
+        MPI_Comm_rank(MPI_COMM_WORLD, &_rank); \
+        if (_rank == 0) { \
+            fprintf(stdout, fmt, ##__VA_ARGS__); \
+        } \
+    } while (0)
+
+#define array_size(_array) (sizeof(_array) / sizeof((_array)[0]))
+
+#define CUDA_CALL(_func) \
+    do { \
+        CUresult __err = (_func); \
+        if (__err != CUDA_SUCCESS) { \
+            const char *__err_str; \
+            cuGetErrorString(__err, &__err_str); \
+            fprintf(stderr, "test_mpi_cuda.c:%-3u %s failed: %d (%s)\n", \
+                    __LINE__, #_func, __err, __err_str); \
+            exit(__err); \
+        } \
+    } while (0)
+
+typedef struct {
+    CUdeviceptr ptr;
+    size_t      size;
+    void        *obj;
+} alloc_mem_t;
+
+typedef struct {
+    const char *name;
+    alloc_mem_t (*alloc)(size_t);
+    void (*free)(alloc_mem_t*);
+} allocator_t;
+
+typedef struct {
+    const char     *description;
+    const unsigned min_gpus;
+    const int      mthread_support;
+    int (*func)(const allocator_t*, const allocator_t*, const void*, size_t,
+                CUdevice);
+} test_t;
+
+typedef struct {
+    CUdeviceptr d_send;
+    CUdeviceptr d_recv;
```

Comment:
minor: maybe use `send_ptr` and `recv_ptr` or use some other consistent names for `CUdeviceptr` variables

Reply:
Renamed the fields.